### PR TITLE
IPC

### PR DESCRIPTION
**Pull request by faboweb**
> Thank you a lot for contributing to Cosmos Voyager!

### Issue

closes: #375

Missing:
- [x] rewrite tests

❤️ Thank you!

### CODE REVIEW

File: app/src/renderer/connectors/node.js [hunk 1]
```diff
@@ -2,17 +2,17 @@
 const RestClient = require('./lcdClient.js')
 const mockedRestClient = require('./lcdClientMock.js')
 const RpcWrapper = require('./rpcWrapper.js')
-const mockedRpcWrapper = require('./rpcWrapperMock.js')
+const MockedRpcWrapper = require('./rpcWrapperMock.js')
```
Comment (jolesbi) [inline app/src/renderer/connectors/node.js]: what is our standard for capital vs camel casing?

File: app/src/renderer/connectors/node.js [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
 
-module.exports = function (nodeIP, relayPort, mocked = false) {
-  const RELAY_SERVER = 'http://localhost:' + relayPort
+module.exports = function (nodeIP, lcdPort, mocked = false) {
+  const LCD_SERVER = 'http://localhost:' + lcdPort
```
Comment (jolesbi) [inline app/src/renderer/connectors/node.js]: 👍

File: app/src/renderer/connectors/node.js [hunk 3]
```diff
@@ -24,7 +24,6 @@ module.exports = function (nodeIP, relayPort, mocked = false) {
   }
   // TODO: eventually, get all data from light-client connection instead of RPC
 
-  connector.setMocked(mocked)
-  connector.initRPC(nodeIP)
```
Comment (jolesbi) [inline app/src/renderer/connectors/node.js]: why don't we need to call initRPC anymore?

Comment (faboweb) [reply app/src/renderer/connectors/node.js]: init is called only after the main process signals, that it has connected to a node

File: app/src/renderer/connectors/node.js (hunk 1 above)
Comment (faboweb) [reply app/src/renderer/connectors/node.js]: I used capital here, as it is a constructor-like function.

File: app/src/renderer/connectors/node.js (hunk 3 above)
Comment (jolesbi) [reply app/src/renderer/connectors/node.js]: is it because we're using IPC instead?

File: app/src/renderer/connectors/node.js (hunk 1 above)
Comment (faboweb) [reply app/src/renderer/connectors/node.js]: In general only classes and constants should have capital letters

File: app/src/renderer/main.js [hunk 4]
```diff
@@ -46,23 +45,9 @@ Vue.use(Router)
...
-
-  node.lcdConnected()
-    .then(connected => {
-      if (connected) {
-        axios.get(`http://localhost:${relayPort}/startsuccess`)
-      }
-    })
+  let lcdPort = getQueryParameter('lcd_port')
+  console.log('Expecting lcd-server on port:', lcdPort)
+  const node = Node('localhost', lcdPort)
```
Comment (jolesbi) [inline app/src/renderer/main.js]: isn't this defaulting to a localhost now instead of the actual node we want to connect to?

Comment (faboweb) [reply app/src/renderer/main.js]: This is confusing I will change this.

File: app/src/renderer/node.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        node.rpc.on('error', () => { })
+
+        node.rpc.ws.destroy()
+      }
+
+      console.log('init rpc with', nodeIP)
+      let newRpc = new RpcClient(`ws://${nodeIP}`)
+      node.rpcOpen = true
+      node.rpcConnecting = false
+      // we need to check immediately if he connection fails. later we will not be able to check this error
```
Comment (jolesbi) [inline app/src/renderer/node.js]: "if the connection"

File: app/src/renderer/node.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+'use strict'
+const { ipcRenderer } = require('electron')
+const RpcClient = require('tendermint')
+const RestClient = require('./lcdClient.js')
+
+module.exports = function (nodeIP, relayPort, lcdPort) {
+  const RELAY_SERVER = 'http://localhost:' + relayPort
```
[the remark this answers is not in the document]
Comment (faboweb) [reply app/src/renderer/node.js]: This was dead code. I deleted the file.

**Pull request merged**